### PR TITLE
changing where require GATK spark settings are applied

### LINKED ISSUE #1096: Spark options aren't being passed to master
Spark parameters set in SparkCommandLineProgram aren't actually being propagated to the spark master.  @LeeTL1220 discovered that this was the case.   The solution is to move them into the SparkContextFactory and apply them to the `SparkConf` instead of the `SparkConfig`

### PR DESCRIPTION
**Pull request by lbergelson**
moving settings from SparkCommandLineProgram to SparkContextFactory
the settings that were being applied to SparkContext.getConf were pointless since getConf is a copy of the configuration
instead they're applied to the SparkConfig in SparkContextFactory.getSparkContext

fixes #1096

[Review on Reviewable](https://reviewable.io/reviews/broadinstitute/gatk/1097)

### CODE REVIEW

**Comment by lbergelson** [comment]
@droazen Responded to your comments.

File: src/main/java/org/broadinstitute/hellbender/engine/spark/SparkContextFactory.java [hunk 1]
```diff
@@ -65,26 +66,31 @@ public void run() {
...
                 .setMaster(master)
                 .set("spark.serializer", KryoSerializer.class.getCanonicalName())
                 .set("spark.kryo.registrator", "org.broadinstitute.hellbender.engine.spark.GATKRegistrator")
-                .set("spark.kryoserializer.buffer.max", "512m");
+                .set("spark.kryoserializer.buffer.max", "512m")
+                .set("spark.driver.maxResultSize", "0")
+                .set("spark.driver.userClassPathFirst", "true")
+                .set("spark.executor.userClassPathFirst", "true")
+                .set("spark.io.compression.codec", "lzf")
+                .set("spark.yarn.executor.memoryOverhead", "600");
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/SparkContextFactory.java]: Create ticket to eventually expose these (either as args in `SparkCommandLineProgram`, or via the `spark-submit`/`gcloud` wrapper that you're writing).

File: src/test/java/org/broadinstitute/hellbender/engine/spark/SparkContextFactoryUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.serializer.KryoSerializer;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class SparkContextFactoryUnitTest extends BaseTest {
+
+    @Test
+    public void testGetSparkContextTest(){
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/engine/spark/SparkContextFactoryUnitTest.java]: Discussed in person ways to make this test more meaningful

File: src/main/java/org/broadinstitute/hellbender/engine/spark/SparkContextFactory.java (hunk 1 above)
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/SparkContextFactory.java]: #1107

**Pull request merged**